lib/elixir/src/elixir_erl_for.erl

```diff
@@ -43,6 +43,18 @@ translate_into(Meta, Cases, Expr, Opts, Return, S) ->
     into -> build_into(Ann, TCases, TExpr, TInto, TUniq, SE)
   end.
 
+validate_uniq(Opts, TInto, _Meta) ->
+  case {lists:keyfind(uniq, 1, Opts), TInto} of
+    {{uniq, true}, false} ->
+      % TODO format this warning properly with Meta
+      io:format(standard_error, "warning: the option :uniq has no effect because the result is not used~n", []),
```

Comment:
I struggled a bit here and couldn't figure it the right way to warn, since `elixir_erl_for` doesn't raise any warning as of today.
I tried to do it based on some other example using `form_warn`, but `E` isn't accessible here.

Reply:
We don't emit warnings from `elixir_erl_*` modules, because they don't have all relevant information. We would need to emit this from `elixir_expand` by porting `translate_block` from `elixir_erl_pass` to `elixir_expand`.

If we do so, we likely want to remove `translate_block` from `elixir_erl_pass`. To do so, we should probably normalize `into` in `elixir_expand` so a comprehension has `:into` set to `[]` if no `:reduce` and no `:into are given and the result is used. If we arrive in `elixir_erl_for` and the comprehension has no `:reduce` and no `:into`, then we assume the result is not used.

Does it make sense?

Reply:
OK, I think I got it working, sorry for the delay!
Is this what you had in mind @josevalim ?